src/gc/env/gcenv.base.h

```diff
@@ -253,12 +253,21 @@ inline uint8_t BitScanForward64(uint32_t *bitIndex, uint64_t mask)
     uint32_t hi = (mask >> 32) & 0xFFFFFFFF;
     uint32_t lo = mask & 0xFFFFFFFF;
     uint32_t fakeBitIndex = 0;
-    if (BitScanForward(&fakeBitIndex, hi))
+    
+    uint8_t result = BitScanForward(bitIndex, lo);
+    if (result != 0)
+    {
+        return result;
+    }
+
+    result = BitScanForward(&fakeBitIndex, hi);
+    if (result != 0)
     {
         *bitIndex = fakeBitIndex + 32;
+        return result;
```

Comment:
You can remove this 'return result' and replace the `return 0` at the end by `return result`. And maybe simplify it even more:
```C
uint8_t result = BitScanForward(bitIndex, lo);
if (result == 0)
{
    result = BitScanForward(&fakeBitIndex, hi);
    if (result != 0)
    {
        *bitIndex = fakeBitIndex + 32;
    }
}

return result;
```

Reply:
Thanks, Jan